include/libtorrent/disk_io_thread.hpp

```diff
@@ -434,6 +435,33 @@ namespace libtorrent
 
 	private:
 
+		friend struct job_queue;
```

Comment:
I could have sworn they weren't, but when I went to find the relevant quote in the spec I discovered that you are right! Checking further, it seems this rule was changed in C++11.